shielded-pool/src/component.rs

```diff
@@ -391,13 +411,15 @@ impl ShieldedPool {
 
     #[instrument(skip(self))]
     async fn write_compactblock_and_nct(&mut self) -> Result<()> {
+        // Extract the compact block, resetting it
+        let compact_block = std::mem::take(&mut self.compact_block);
+        let height = compact_block.height;
```

Comment:
We should use the `get_block_height` method here, since that's where the actual bug was -- this is still dependent on someone not refactoring this to reorder changes to the compact block with grabbing the height, or grabbing the height off of `self.compact_block` instead.

Reply:
Right! Done.